js_utils/text_validation.js

```diff
@@ -186,6 +187,13 @@ export class TextValidation {
     if (barcode_header !== "ML" && barcode_header !== "MS") {
       return " ";
     }
+    // check barcode matches type
+    if (
+      (barcode && type === "plate_barcode" && !barcode.includes("ML")) ||
+      (barcode && type === "stim_barcode" && !barcode.includes("MS"))
+    ) {
+      return "barcode type no match";
```

Comment:
I'm not sure what all uses this function, but just to be safe I would return `" "` like every other failure

Reply:
Also this check needs to be done on `barcode_header` here. You can modify the if cond above on line 187 since it's redundant now